geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/iterators/ArrowScan.scala

```diff
@@ -272,12 +277,12 @@ object ArrowScan extends LazyLogging {
       SimpleFeatureTypes.createType(options(SftKey), options(SpecKey))
 
     def encoding(e: SimpleFeatureEncoding): (String, String) =
-      EncodingKey -> s"${e.fids.getOrElse("")}:${e.geometry}:${e.date}"
+      EncodingKey -> s"${e.fids.getOrElse("")}:${e.geometry}:${e.date}:${e.axisOrder}"
 
     def encoding(options: Map[String, String]): SimpleFeatureEncoding = {
-      val Array(fids, geom, dtg) = options(EncodingKey).split(":")
+      val Array(fids, geom, dtg, axisOrder) = options(EncodingKey).split(":")
```

Comment:
for iterator code, we want to ensure back-compatibility - so that means handling the case where axis order is not set. the cleanest thing would probably be to put axis order in a separate key.